Library/Homebrew/download_strategy.rb

```diff
@@ -447,7 +447,12 @@ def _fetch
 # Useful for installing jars.
 class NoUnzipCurlDownloadStrategy < CurlDownloadStrategy
   def stage
-    cp cached_location, basename_without_params, preserve: true
+    # Because we create & set HOME as (build|test)path/../brew_home we need to
+    # ensure everything is staged at least one subdirectory deep, otherwise
+    # brew_home ends up in HOMEBREW_TEMP rather than the (build|test)path.
+    mkdir "staging"
+    cp cached_location, "staging/#{basename_without_params}", preserve: true
+    cd "staging"
```

Comment:
Do other strategies do the `cd`?

Reply:
_Inside_ the buildpath/testpath itself? _I think this might_ be unique. I'm essentially doing it this way to preserve backwards compatibility with formulae without requiring changes on that side of things for this aspect, whilst still allowing us to force everything once subdirectory deep.

Bit of a dance maybe but so far what I've tested works as expected.

Reply:
not unique. idiomatic way is `chdir`
See https://github.com/Homebrew/brew/blob/master/Library/Homebrew/download_strategy.rb#L250 and https://github.com/Homebrew/brew/blob/master/Library/Homebrew/download_strategy.rb#L269-L279

Reply:
Cool, thanks both 👍

Reply:
> idiomatic way is chdir

Ah, of course 👍. That automatic jump didn't seem to work for `nounzip`, but that seems to be expected; we'd presumably need a `when 2 then` but don't know if you're happy with where the logic is now or whether you want a more general case.

Reply:
maybe a function for handling nounzip and tarball-extracting-to-current-dir and whatever else comes up?

Reply:
if child count is 0, empty archive
elsif child count is 1 and child is a dir, cd into dir
elsif one of the children is named staging already, set the house on fire
else
mkdir staging and move children - staging into staging and cd into staging

Reply:
(could probably live in chdir or be called from there)

Reply:
Yeah, that sounds like a viable option, I'll look into it.